packages/account/src/providers/fuel-graphql-subscriber.ts

```diff
@@ -10,14 +10,12 @@ type FuelGraphQLSubscriberOptions = {
 };
 
 export class FuelGraphqlSubscriber implements AsyncIterator<unknown> {
-  private stream!: ReadableStreamDefaultReader<Uint8Array>;
   private static textDecoder = new TextDecoder();
 
-  public constructor(private options: FuelGraphQLSubscriberOptions) {}
-
-  private async setStream() {
-    const { url, query, variables, fetchFn } = this.options;
+  private constructor(private stream: ReadableStreamDefaultReader<Uint8Array>) {}
 
+  public static async create(options: FuelGraphQLSubscriberOptions) {
```

Comment:
I moved `FuelGraphqlSubcriber` instantiation to be async as a preparatory step for #2936. I tried introducing the solution for that issue here in commit https://github.com/FuelLabs/fuels-ts/pull/2962/commits/80511131b68754288f7e686fc040413e8ea14141 but I reverted it because the CI disagreed that it's a solution :sweat_smile:. 

This change is necessary for the usage of `submitAndAwait` endpoint because the transaction can fail when submitted, but the failure isn't manifested until a call to the `next` method is done. I will add this failure handling in the subsequent PR that will tackle #2936, after I sync with the `fuel-core` team. We currently aren't using `submitAndAwait` anywhere so it's not a problem to leave it as-is.